frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -415,12 +418,15 @@ function getStackedDataLabelTransform(
   };
 }
 
-function filterNullDimensionValues(dataset: ChartDataset) {
-  // TODO show warning message
+export function filterNullDimensionValues(
+  dataset: ChartDataset,
+  showWarning?: ShowWarning,
+) {
   const filteredDataset: ChartDataset = [];
 
   dataset.forEach((datum, originalIndex) => {
     if (datum[X_AXIS_DATA_KEY] == null) {
+      showWarning?.(nullDimensionWarning().text);
```

Comment:
Moving here logic from https://github.com/metabase/metabase/pull/40763 which warns while sorting may be the way to reduce the surface and put similar warnings related to x-axis values together